lib/metrics.py

```diff
@@ -179,10 +179,10 @@ def check_metric(**params):
     assert 1.0 >= ssim[2] >= minv
 
   elif "md5" == type:
-    numbytes = get_framesize(
+    numbytes = params.get("numbytes", get_framesize(
```

Comment:
`numbytes` should be retrieved from `metric = params["metric"]` sub-entry.

That is, the user config should specify in test case spec, like:
```
metric = dict(type = "md5", numbytes = -1)
```

Also, if user is allowed to set numbytes now, then it doesn't always have to be "whole file" mode... so please update commit message accordingly.  For example, "metrics: support user-specified md5 numbytes" and explain in the message body what behavior is expected based on user and default values.